### ISSUE TITLE
About the experimental results

### ISSUE THREAD

**Issue by ghost** [FlappyPeggy/DMAD#6]
hello, when I use the default configuration, the average image AUROC result is only 98.2, do you know why?

class | image auroc
————————
carpet | 96.9
grid | 96.8
leather | 100
tile | 98.8
wood | 99.6
bottle | 100
cable | 97.8
capsule | 96.5
hazelnut | 100
metal_nut | 99.9
pill | 96.1
screw | 93.5
toothbrush | 100
transistor | 98.9
zipper | 98.7
——————
mean |98.2

**Comment by FlappyPeggy** [comment]
*Reasons:*

1. Distance-based prototypical learning is unstable and increases variance despite improving the best performance. 
2. PPDM introduces slight shortcut learning compared to PDM as discussed in the paper (which breaks the third precondition). That is, the pre-deformation module may be used to eliminate prototype information.

*Solutions:*

0. Log each evaluation to check best performance;
1. Manually set $self.vq = False$ in $resnet.py$ to disable the prototypical learning (in this case, set $epoch=200$);
2. Enable the comments in $Train mvtec.py$ *L126-127*, use early-stop to prevent $VQ Loss$ from enhancing shortcut-learning (you can modify the number ``200'' with supervising the main loss and  regularization term instead of optimizing $VQ Loss$);
3. Reduce the weight of $VQ Loss$;
4. Use hand-crafted pseudo anomalies as the validation set to obtain the optimal early-stop epoch and hyperparameters;
